src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -55,7 +55,7 @@ public class DefaultHtmlGenerator : IHtmlGenerator
             _htmlEncoder = htmlEncoder;
 
             // Underscores are fine characters in id's.
-            IdAttributeDotReplacement = "_";
+            IdAttributeDotReplacement = optionsAccessor.Options.HtmlHelperOptions.IdAttributeDotReplacement;
```

Comment:
Makes some sense but leaves this setting in a slightly strange place.  `DefaultHtmlGenerator` owns an `IdAttributeDotReplacement` value that newly-created instances will not inherit.  `HtmlHelper` also has a `public` `IdAttributeDotReplacement` that reads from / writes to its `private` `IHtmlGenerator` instance .

MVC 5 equivalent of this property is `static`. It is _not_ possible there to give it a setting unique to any scope of `HtmlHelper` instance.

One possibility: Remove the `IHtmlHelper` and `IHtmlGenerator` setters for this property.  Then `MvcOptionsSetup` would be the _single_ way to change `IdAttributeDotReplacement`.  @Eilon thoughts?